### DISCUSSION TITLE
Is there a way to stop the mysql service from the command line?

### DISCUSSION THREAD

**Discussion by sistemanick** [leokhoa/laragon#886] [Q&A] [answered]
I have a backup .bat script that runs every day but it'll fail to pick up the mysql data files if mysql is running. How do I stop mysql from the command line and restart it when the backup is done?

**Comment by ppfoong** [comment]
There is a batch file here for you to refer to:
https://github.com/leokhoa/laragon/issues/790

**Reply by sistemanick** [reply]
> There is a batch file here for you to refer to: #790

`net stop mysql` doesn't work at all. It just gives the error "The service name is invalid.".

I ran `sc query | findstr /i "mysql"` and found nothing.

**Reply by ppfoong** [reply, marked as answer]
What is the outcome running this command?
`taskkill /F /IM mysqld.exe`

By the way, proper way of backing up MySQL database is to dump the data into .sql file, and to backup that dump file.

**Reply by sistemanick** [reply]
> What is the outcome running this command? `taskkill /F /IM mysqld.exe`
> 
> By the way, proper way of backing up MySQL database is to dump the data into .sql file, and to backup that dump file.

Okay I've updated script to do a mysqldump before backing up the resulting .sql. Thanks for the tip.
